app/routes/public/sessions/list.js

```diff
@@ -16,169 +16,175 @@ export default Route.extend({
+              }
+            },
+            {
+              name : 'starts-at',
+              op   : 'ge',
+              val  : moment().startOf('month').toISOString()
+            },
+            {
+              name : 'starts-at',
+              op   : 'lt',
+              val  : moment().add('month').toISOString()
+            },
+            {
+              or: [
+                {
+                  name : 'state',
                   op   : 'eq',
-                  val  : eventDetails.id
+                  val  : 'confirmed'
+                },
+                {
+                  name : 'state',
+                  op   : 'eq',
+                  val  : 'accepted'
                 }
-              },
-              {
-                name : 'starts-at',
-                op   : 'ge',
-                val  : moment().startOf('month').toISOString()
-              },
-              {
-                name : 'starts-at',
-                op   : 'lt',
-                val  : moment().add('month').toISOString()
-              },
-              {
-                or: [
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'confirmed'
-                  },
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'accepted'
-                  }
-                ]
-              }
-            ]
-          }
-        ]
-      });
+              ]
+            }
+          ]
+        }
+      ];
+
     } else {
-      sessions = await this.store.query('session', {
-        filter: [
-          {
-            and: [
-              {
-                name : 'event',
-                op   : 'has',
-                val  : {
-                  name : 'identifier',
+
+      filterOptions = [
+        {
+          and: [
+            {
+              name : 'event',
+              op   : 'has',
+              val  : {
+                name : 'identifier',
+                op   : 'eq',
+                val  : eventDetails.id
+              }
+            },
+            {
+              or: [
+                {
+                  name : 'state',
                   op   : 'eq',
-                  val  : eventDetails.id
+                  val  : 'confirmed'
+                },
+                {
+                  name : 'state',
+                  op   : 'eq',
+                  val  : 'accepted'
                 }
-              },
-              {
-                or: [
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'confirmed'
-                  },
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'accepted'
-                  }
-                ]
-              }
-            ]
-          }
-        ]
-      });
+              ]
+            }
+          ]
+        }
+      ];
```

Comment:
Massive duplication. All filters are almost same

Reply:
They were implemented like this before, Also month was not working. 
Fixed them now